C/c4Index.cc

```diff
@@ -65,20 +65,39 @@ C4IndexUpdater::C4IndexUpdater(Retained<litecore::LazyIndexUpdate> u, C4Collecti
 
 C4IndexUpdater::~C4IndexUpdater() = default;
 
-size_t C4IndexUpdater::count() const { return _update->count(); }
+size_t C4IndexUpdater::count() const {
+    if ( hasFinished() ) {
+        WarnError("C4IndexUpdater::count() called on finished updater.");
+        return 0;
+    }
+    return _update->count();
+}
 
-FLValue C4IndexUpdater::valueAt(size_t i) const { return _update->valueAt(i); }
+FLValue C4IndexUpdater::valueAt(size_t i) const {
+    if ( hasFinished() ) {
+        litecore::error::_throw(litecore::error::NotOpen, "C4IndexUpdater::valueAt() called on finished updater.");
+    }
+    return _update->valueAt(i);
+}
 
 void C4IndexUpdater::setVectorAt(size_t i, const float* vector, size_t dimension) {
+    if ( hasFinished() ) {
+        litecore::error::_throw(litecore::error::NotOpen, "C4IndexUpdater::setVectorAt() called on finished updater.");
```

Comment:
There is not much need to throw an exception here.  Just log and return.  An exception provides no value and wastes cycles with try / catch.

Reply:
This is actually sneaky. 
```
bool c4indexupdater_skipVectorAt(C4IndexUpdater* update, size_t i) noexcept {
    return tryCatch(nullptr, [&] { update->skipVectorAt(i); });
}
```
It relies on tryCatch to return boolean.

Reply:
This makes this API inconsistent with skipVector....as this will return an error while skipVector will not.  I don't mind these calls succeeding as no-ops in this case since the eventual call to finish will alert the user to their mistake.